bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/channelhelper/EventChannelHelper.java

```diff
@@ -0,0 +1,95 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.netatmo.internal.api.dto.Event;
+import org.openhab.binding.netatmo.internal.api.dto.HomeEvent;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+
+/**
+ * The {@link EventChannelHelper} handles specific channels of cameras
+ *
+ * @author Gaël L'hopital - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class EventChannelHelper extends ChannelHelper {
+    private boolean isLocal;
+    private @Nullable String vpnUrl;
+    private @Nullable String localUrl;
+    private boolean isMonitoring;
+
+    public EventChannelHelper() {
+        this(GROUP_LAST_EVENT);
+    }
+
+    protected EventChannelHelper(String groupName) {
+        super(groupName);
+    }
+
+    public void setUrls(String vpnUrl, @Nullable String localUrl, boolean isMonitoring) {
+        this.localUrl = localUrl;
+        this.vpnUrl = vpnUrl;
+        this.isLocal = localUrl != null;
+        this.isMonitoring = isMonitoring;
+    }
+
+    @Override
+    protected @Nullable State internalGetEvent(String channelId, Event event) {
+        switch (channelId) {
+            case CHANNEL_EVENT_TYPE:
+                return toStringType(event.getEventType());
+            case CHANNEL_EVENT_MESSAGE:
+                return toStringType(event.getName());
+            case CHANNEL_EVENT_TIME:
+                return new DateTimeType(event.getTime());
+            case CHANNEL_EVENT_PERSON_ID:
+                return toStringType(event.getPersonId());
+            case CHANNEL_EVENT_CAMERA_ID:
+                return toStringType(event.getCameraId());
+            case CHANNEL_EVENT_SUBTYPE:
+                return event.getSubTypeDescription().map(d -> toStringType(d)).orElse(UnDefType.NULL);
+            case CHANNEL_EVENT_SNAPSHOT:
+                return toRawType(event.getSnapshotUrl());
+            case CHANNEL_EVENT_SNAPSHOT_URL:
+                return toStringType(event.getSnapshotUrl());
+        }
+        if (event instanceof HomeEvent) {
+            HomeEvent homeEvent = (HomeEvent) event;
+            switch (channelId) {
+                case CHANNEL_EVENT_VIDEO_STATUS:
+                    return homeEvent.getVideoId() != null ? toStringType(homeEvent.getVideoStatus()) : UnDefType.NULL;
+                case CHANNEL_EVENT_VIDEO_LOCAL_URL:
+                    return getStreamURL(true, homeEvent.getVideoId());
+                case CHANNEL_EVENT_VIDEO_VPN_URL:
+                    return getStreamURL(false, homeEvent.getVideoId());
+            }
+        }
+        return null;
+    }
+
+    private State getStreamURL(boolean local, @Nullable String videoId) {
+        String url = local ? localUrl : vpnUrl;
+        if ((local && !isLocal) || url == null || videoId == null || !isMonitoring) {
```

Comment:
isMonitoring should be ignored.
You can access a recorded video event even if the monitoring status is disabled.

Reply:
Fixed in my submitted PR

Reply:
Correct, and you did it. Thanks.